### PR TITLE
Added BUFFER_LENGTH in I2Cdev.h

### PR DESCRIPTION
**Pull request by 0xfabi**
BUFFER_LENGTH has to be declared for functions like readByte or readWord.

### CODE REVIEW

**Comment by jrowberg** [comment]
The `BUFFER_LENGTH` constant is part of some built-in libraries, and should only be defined if it isn't already (although in that case, defining it inside `I2Cdev.*` is probably unsafe since it's supposed to come from outside of the library). I don't think this is a safe change as-is.

**Comment by ZurMaD** [comment]
BUFFER_LENGTH has conflicts, please add #define BUFFER_LENGTH 32

**Comment by ZurMaD** [comment]
Your library is use in Particle Photon company and make errors because BUFFER_LENGHT was not declarated in the scope.

**Comment by arvidtp** [comment]
I believe ideally this should be fixed in the Wire.h file in whichever non-AVR arduino hardware cores where it is not defined.

For instance, I ran across this issue with [arduino-esp32](https://github.com/espressif/arduino-esp32) on the Sparkfun ESP32 Thing board. Arduino-esp32's Wire.h file uses I2C_BUFFER_LENGTH instead of BUFFER_LENGTH. So appending the line

`#define BUFFER_LENGTH I2C_BUFFER_LENGTH`

would fix it.

But barring that, I2Cdev.h could do something like

```
#if I2CDEV_IMPLEMENTATION == I2CDEV_ARDUINO_WIRE
        #include <Wire.h>
        #ifndef BUFFER_LENGTH
                #warning "i2c BUFFER_LENGTH not defined. Check the Wire.h file for you hardware core. Self-defining as 32."
        	#define BUFFER_LENGTH 32
        #endif
    #endif
```

File: Arduino/I2Cdev/I2Cdev.h [hunk 1]
```diff
@@ -68,6 +68,7 @@ THE SOFTWARE.
 #define I2CDEV_BUILTIN_FASTWIRE     3 // FastWire object from Francesco Ferrara's project
 #define I2CDEV_I2CMASTER_LIBRARY    4 // I2C object from DSSCircuits I2C-Master Library at https://github.com/DSSCircuits/I2C-Master-Library
 
+#define BUFFER_LENGTH 32
```
Comment (adamrabbani) [inline Arduino/I2Cdev/I2Cdev.h]: yes, this is must added for esp32 with library i2cdevlib

**Comment by jrowberg** [comment]
Fixed already with a cross-platform compiler macro section in `I2Cdev.h`, added recently. Thanks!

**Pull request closed without merging**